wgpu/src/util/texture_blitter.rs

```diff
@@ -0,0 +1,168 @@
+use wgt::BlendState;
+
+use crate::{
+    include_wgsl, AddressMode, BindGroupDescriptor, BindGroupEntry, BindGroupLayout,
+    BindGroupLayoutDescriptor, BindGroupLayoutEntry, BindingType, ColorTargetState, ColorWrites,
+    CommandEncoder, Device, FilterMode, FragmentState, FrontFace, LoadOp, MultisampleState,
+    PipelineCompilationOptions, PipelineLayoutDescriptor, PrimitiveState, PrimitiveTopology,
+    RenderPassDescriptor, RenderPipeline, RenderPipelineDescriptor, Sampler, SamplerBindingType,
+    SamplerDescriptor, ShaderStages, StoreOp, TextureFormat, TextureSampleType, TextureView,
+    TextureViewDimension, VertexState,
+};
+
+/// A texture blitting utility
+/// Use this if you want to just render/copy texture A to texture B where [CommandEncoder::copy_texture_to_texture] would not work because of either
+/// - Textures are in incompatible formats
+/// - Textures are of different sizes
```

Comment:
```suggestion
/// Texture Blitting (Copying) Utility
///
/// Use this if you want to just render/copy texture A to texture B where [`CommandEncoder::copy_texture_to_texture`] would not work because:
/// - Textures are in incompatible formats.
/// - Textures are of different sizes.
/// - Your copy destination is the surface texture and does not have the `COPY_DST` usage.
```